exercises/gagik_vardanyan/ex02_oo_basics/src/include/ShapeSorter.h

```diff
@@ -12,14 +12,14 @@
 
 class ShapeSorter {
 private:
-    std::vector<Shape> & shapes; /**< vector of shapes */
+    std::vector<Shape*> & shapes; /**< vector of shape pointers */
```

Comment:
Good. You do indeed need to use a pointer here. If you had c++11 available, I would suggest using a scoped manage smart pointer of some kind rather than storing the raw pointer directly, but this will do fine.